tensorflow/core/lib/gif/gif_io.cc

```diff
@@ -78,11 +78,14 @@ uint8* Decode(const void* srcdata, int datasize,
   if (DGifSlurp(gif_file) != GIF_OK) {
     *error_string = absl::StrCat("failed to slurp gif file: ",
                                  GifErrorStringNonNull(gif_file->Error));
-    // Only stop load if no images are detected.
-    if (gif_file->ImageCount <= 0) {
-      return nullptr;
+    // Stop load if no images are detected or the allocation of the last image
+    // buffer was failed.
+    if (gif_file->ImageCount <= 0 ||
+        gif_file->SavedImages[gif_file->ImageCount - 1].RasterBits == NULL) {
```

Comment:
I think the `return nullptr` is supposed to go inside these brackets.  Otherwise, this looks like a noop.

Reply:
Oops, my fault
Fixed